src/contexts/events.tsx

```diff
@@ -153,6 +154,14 @@ const useEvents = () => {
     trackedEventTypes.current = new Set();
   }, []);
 
+  useEffect(() => {
```

Comment:
Well in that case, it would send a `connect` event but with no address.  
I think it would be better to have the update logic only on one `useEffect` if possible, otherwise it's hard to follow. 

And in fact it could lead to deadlock (maybe not the case here).

Reply:
True, I missed that. I changed it again and I think it's fixed now.